lstchain/image/muon/muon_analysis.py

```diff
@@ -410,29 +427,38 @@ def create_muon_table():
 
     output_parameters['event_id'].append(event_id)
     output_parameters['event_time'].append(event_time)
-    output_parameters['ring_size'].append(muonintensityparam.ring_size)
+    output_parameters['ring_size'].append(size)
     output_parameters['size_outside'].append(size_outside_ring)
-    output_parameters['ring_center_x'].append(muonringparam.ring_center_x.value)
-    output_parameters['ring_center_y'].append(muonringparam.ring_center_y.value)
-    output_parameters['ring_radius'].append(muonringparam.ring_radius.value)
-    output_parameters['ring_width'].append(muonintensityparam.ring_width.value)
+    output_parameters['ring_center_x'].append(muonringparam.center_x.value)
+    output_parameters['ring_center_y'].append(muonringparam.center_y.value)
+    output_parameters['ring_radius'].append(muonringparam.radius.value)
+    output_parameters['ring_width'].append(muonintensityparam.width.value)
     output_parameters['good_ring'].append(good_ring)
-    output_parameters['muon_efficiency'].append(muonintensityparam.optical_efficiency_muon)
-    output_parameters['ring_containment'].append(muonringparam.ring_containment)
-    output_parameters['ring_completeness'].append(muonintensityparam.ring_completeness)
-    output_parameters['ring_pixel_completeness'].append(muonintensityparam.ring_pix_completeness)
-    output_parameters['impact_parameter'].append(muonintensityparam.impact_parameter.value)
-    output_parameters['impact_x_array'].append(muonintensityparam.impact_parameter_pos_x.value)
-    output_parameters['impact_y_array'].append(muonintensityparam.impact_parameter_pos_y.value)
+
+    # WARNING!! BELOW WE INTRODUCE A FACTOR 2 CORRECTION BECAUSE OF AN ERROR
+    # IN ctapipe 0.8, see https://github.com/cta-observatory/ctapipe/issues/1377
+    # THIS HAS TO BE REMOVED AS SOON AS IT IS FIXED IN A NEW VERSION OF CTAPIPE!
+    output_parameters['muon_efficiency'].append(
+            2.0 * muonintensityparam.optical_efficiency)
+
+    output_parameters['ring_containment'].append(muonparameters.containment)
+    output_parameters['ring_completeness'].append(muonparameters.completeness)
+    #  missing in ctapipe 0.8:
+    # output_parameters['ring_pixel_completeness'].append(muonintensityparam.ring_pix_completeness)
+    output_parameters['impact_parameter'].append(muonintensityparam.impact.value)
+    output_parameters['impact_x_array'].append(muonintensityparam.impact_x.value)
+    output_parameters['impact_y_array'].append(muonintensityparam.impact_y.value)
     output_parameters['radial_stdev'].append(radial_distribution['standard_dev'].value)
     output_parameters['radial_skewness'].append(radial_distribution['skewness'])
     output_parameters['radial_excess_kurtosis'].append(radial_distribution['excess_kurtosis'])
-    output_parameters['num_pixels_in_ring'].append(np.sum(muonintensityparam.mask))
+    #  missing in ctapipe 0.8:
+    #output_parameters['num_pixels_in_ring'].append(np.sum(muonintensityparam.mask))
```

Comment:
and this one? was it used?

Reply:
No, and anyway we cannot use the mask now to do the efficiency fit, so this is something to be modified when the needed fix in ctapipe is made (or we do it provisionally in lstchain by copying the corresponding class).